### PR TITLE
Add statenotifier (Create StateNotifier template)

### PR DESCRIPTION
**Pull request by HugoHeneault**
It isn't from Riverpod but I use this one quite often and I think it is missing here :-)

### CODE REVIEW

**Comment by HugoHeneault** [comment]
Or maybe we can update `StateNotifierProvider` snippet to create the StateNotifier and its Provider?

```
class $name$Notifier extends StateNotifier<$result$> {
  $name$Notifier(): super($initialState$);

  $END$
}

final $name$Provider = StateNotifierProvider<$name$Notifier>((ref) => $name$Notifier());

```

**Comment by tbm98** [comment]
I think we should just create each of them independently.

File: src/main/resources/liveTemplates/Flutter_riverpod_snippets.xml [hunk 1]
```diff
@@ -69,6 +69,15 @@
       <option name="DART_TOPLEVEL" value="true" />
     </context>
   </template>
+  <template name="statenotifier" value="class $name$Notifier extends StateNotifier&lt;$result$&gt; {&#10;  $name$Notifier(): super($initialState$);&#10;&#10;  $END$&#10;}" description="Create StateNotifier template" toReformat="false" toShortenFQNames="true">
+    <variable name="name" expression="" defaultValue="" alwaysStopAt="true" />
+    <variable name="result" expression="" defaultValue="" alwaysStopAt="true" />
+    <variable name="initialState" expression="" defaultValue="" alwaysStopAt="true" />
+    <context>
+      <option name="DART" value="true" />
```
Comment (tbm98) [inline src/main/resources/liveTemplates/Flutter_riverpod_snippets.xml:77]: It should be set to `DART_TOPLEVEL` scope.

**Review by tbm98 — CHANGES_REQUESTED** [review]

**Comment by HugoHeneault** [comment]
Also would you add the `hacktoberfest` topic to this repo saying that you allows other devs to contribute to your repo while [participating](https://hacktoberfest.digitalocean.com/) to it?

Thanks :)

**Review by tbm98 — APPROVED** [review]
Great!

**Comment by tbm98** [comment]
> Also would you add the `hacktoberfest` topic to this repo saying that you allows other devs to contribute to your repo while [participating](https://hacktoberfest.digitalocean.com/) to it?

How to apply it ?

**Comment by HugoHeneault** [comment]
> > Also would you add the `hacktoberfest` topic to this repo saying that you allows other devs to contribute to your repo while [participating](https://hacktoberfest.digitalocean.com/) to it?
> 
> How to apply it ?

They say "Maintainers can opt-in to participate by classifying their projects with the `hacktoberfest` topic.", it's on the main page in the about section:

image

**Comment by tbm98** [comment]
Yes. I will try it 😁

Vào 19:30, Th 7, 10 thg 10, 2020 Hugo Heneault <notifications@github.com>
đã viết:

**Pull request merged**